src/commonMethods/commonFunctions.js

```diff
@@ -79,16 +173,318 @@ export const calcAvgFPS = function() {
   average = average.toFixed(2)
   this.$data.write('average', average)
 }
+
+/**
+ * This function is used to get screenshot
+ * @returns {Promise<void>}
+ */
+export const screenshot = async function() {
+  let url = `http://${constants.host}:80/Service/Snapshot/Capture?${moment().valueOf()}`
+  // create a new promise inside of the async function
+  let bufferData = new Promise((resolve, reject) => {
+    _http
+      .get(url, function(res) {
+        if (res.headers['content-length'] === undefined)
+          this.$log(
+            'Framework did not return a content-length! This will slow down the screenshot module.'
+          )
+        var buffers = []
+        var imageSize = res.headers['content-length']
+
+        res.on('data', function(chunk) {
+          buffers.push(Buffer.from(chunk))
+        })
+
+        res.on('end', function() {
+          return resolve(Buffer.concat(buffers, parseInt(imageSize)))
+        })
+      })
+      .on('error', function(e) {
+        return reject(e)
+      })
+  })
+
+  // wait for the promise to resolve
+  let result = await bufferData
+  this.$data.write('screenshotResult', result)
+}
+/**
+ * This function is used to kill the process
+ * @param process
+ */
+export const killProcess = function(process) {
+  exec({ cmd: `killall -9 ${process}` }, err => {
+    return true
+  })
+}
+
+/**
+ * This function is used to stop the process
+ * @param process
+ */
+export const stopProcess = function(process) {
+  return exec({ cmd: `killall ${process}` }, () => {
+    return true
+  })
+}
+export const startFramework = function() {
+  return exec({ cmd: 'nohup WPEFramework WPEProcess -b &', cbWhenStarted: true })
+}
+
+export const startHttpServer = function(requestFunction) {
```

Comment:
Serving up content through a local webserver won't work in a browser environment. Probably we should move the content served up like this to an online location (CDN or so). @wouterlucas